---
layout: workshop_main_7day
permalink: /CSHL_2019
title: HT-Bio
header1: Workshop Pages for Students
header2: High-throughput Biology - From Sequence to Networks 2019
image: /site_images/CBW-CSHL-graphic-square.png
keywords: Visualize genomic data, pathway analysis, integrate clinical data
description: Course covers the key bioinformatics concepts and tools required to analyze  genomic data sets and access and work with data sets in the cloud. 
instructors: Sara Goodwin, Jared Simpson, Mathieu Bourgey, Zhibin Lu, Heather Gibling, Malachi Griffith, Obi Griffith, Peter Ronning, Kelsy Cotto, Juri Reimand, Robin Haw, Veronique Voisin, Quaid Morris, Michael Hoffman
length: 7 days
---

# Welcome <a id="welcome"></a>

Welcome to High-throughput Biology - From Sequence to Networks 2019.  

The course schedule can be found [here](https://bioinformaticsdotca.github.io/CSHL_2019_schedule)  

If you have questions or comments, visit the workshop [Q&A forum](https://classroom.google.com). You may need to join the group by clicking on the "+" symbol and entering the class code.

You can download the printed course manual [here](https://drive.google.com/a/bioinformatics.ca/file/d/1Sjyz6unTX1T4Fed98Rf1WdiM8xe-azwD/view?usp=sharing).  

***

# Class Photo

Coming soon! 

***

# Pre-Workshop Materials <a id="preworkshop"></a>

## Laptop Setup Instructions

Instructions to setup your laptop can be found [here](https://bioinformaticsdotca.github.io/CSHL_2019_laptop_setup).

## Pre-workshop Tutorials

1) **R Preparation tutorials**: You are expected to have completed the following tutorials in **R** beforehand. The tutorial should be very accessible even if you have never used **R** before.

* The [CBW R tutorial](http://bioinformatics-ca.github.io/CBW_R_Tutorial/)
* [Quick and Dirty Guide to **R**](http://ww2.coastal.edu/kingw/statistics/R-tutorials/text/quick&dirty_R.txt)  
* The [R Tutorial](http://www.cyclismo.org/tutorial/R/) up to and including 5. Basic Plots
* The [R command cheat sheet](../../resources/R_Short-refcard.pdf)

2) **UNIX Preparation tutorials**:  

* [UNIX Bootcamp](https://rnabio.org/module-00-setup/0000/06/01/Unix/)
* Tutorials #1-3 on [UNIX Tutorial for Beginners](http://www.ee.surrey.ac.uk/Teaching/Unix/)  
* [Unix Cheat sheet](http://www.rain.org/~mkummel/unix.html) 

3) [Sequencing Terminology](http://www.ncbi.nlm.nih.gov/projects/genome/glossary.shtml)

4)  **Cytoscape Preparation tutorials**: Complete the [introductory tutorial to Cytoscape](http://opentutorials.cgl.ucsf.edu/index.php/Portal:Cytoscape3)

* Introduction to Cytoscape3 - User Interface

* Introduction to Cytoscape3 - Welcome Screen

* Filtering and Editing in Cytoscape 3

## Pre-workshop Readings  

Before coming to the workshop, read these:
  
  * [Integrative Genomics Viewer (IGV): high-performance genomics data visualization and exploration](http://www.ncbi.nlm.nih.gov/pubmed/22517427)
  
  * [Genome structural variation discovery and genotyping](http://www.ncbi.nlm.nih.gov/pubmed/21358748)
  
  * [A survey of sequence alignment algorithms for next-generation sequencing](http://www.ncbi.nlm.nih.gov/pubmed/20460430)
  
  * [Genotype and SNP calling from next-generation sequencing data](http://www.ncbi.nlm.nih.gov/pubmed/21587300)
  
  * [Informatics for RNA Sequencing: A Web Resource for Analysis on the Cloud](https://www.ncbi.nlm.nih.gov/pubmed/26248053)  

  * [Transcript-level expression analysis of RNA-seq experiments with HISAT, StringTie and Ballgown](https://www.ncbi.nlm.nih.gov/pubmed/27560171)
  
  * [ENCODE RNA-Seq Standards](https://genome.ucsc.edu/ENCODE/protocols/dataStandards/ENCODE_RNAseq_Standards_V1.0.pdf)
  
  * [Methods to study splicing from high-throughput RNA sequencing data](http://www.ncbi.nlm.nih.gov/pubmed/24549677)  
  
  * [A comprehensive assessment of RNA-seq accuracy, reproducibility and information content by the Sequencing Quality Control Consortium](http://www.ncbi.nlm.nih.gov/pubmed/25150838)

  * [De novo transcript sequence reconstruction from RNA-seq using the Trinity platform for reference generation and analysis](http://www.nature.com/nprot/journal/v8/n8/full/nprot.2013.084.html)
  
  * [Recurrent chimeric RNAs enriched in human prostate cancer identified by deep sequencing](http://www.ncbi.nlm.nih.gov/pubmed/21571633)
  
  * [Pathway enrichment analysis and visualization of omics data using g:Profiler, GSEA, Cytoscape and EnrichmentMap](https://www.nature.com/articles/s41596-018-0103-9)
  
  * [iRegulon: From a Gene List to a Gene Regulatory Network Using Large Motif and Track Collections](http://www.ncbi.nlm.nih.gov/pubmed/25058159)
  
  * [The GeneMANIA prediction server: biological network integration for gene prioritization and predicting gene function](http://www.ncbi.nlm.nih.gov/pubmed/20576703)
  
  * [GeneMANIA Prediction Server 2013 Update](http://www.ncbi.nlm.nih.gov/pubmed/23794635)
  
  * [How to visually interpret biological data using networks](http://www.ncbi.nlm.nih.gov/pubmed/19816451)
  
  * [g:Profiler--a web-based toolset for functional profiling of gene lists from large-scale experiments](http://www.ncbi.nlm.nih.gov/pubmed/17478515)
  
  * [g:Profiler--a web server for functional interpretation of gene lists (2011 update)](http://www.ncbi.nlm.nih.gov/pubmed/21646343)
  
  * [Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles](http://www.ncbi.nlm.nih.gov/pubmed/16199517)
  
  * [Expression data analysis with reactome](http://www.ncbi.nlm.nih.gov/pubmed/25754994)  

***

# Day 1 <a id="day1"></a>

## Welcome

*<font color="#827e9c">Michelle Brazas</font>*

## Module 1: Introduction to High-throughput Sequencing

*<font color="#827e9c">Sara Goodwin</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1rLdASUMVirxbNgI_EzcUNwXHbrVh82qB/view?usp=sharing)

## Module 2: Data Visualization

*<font color="#827e9c">Heather Gibling</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1CYJpWRQI4wG1BkhHFialbxYibZm4yQK_/view?usp=sharing)

[Lab Practical](https://bioinformaticsdotca.github.io/CSHL_2019_Module2_lab)

## Module 3: Genome Alignment

*<font color="#827e9c">Mathieu Bourgey</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1WVZ8_tIsrH8v86YlL_JR6uUZmf6JsyiV/view?usp=sharing)

[Lab Practical: Connecting to the Cloud](https://drive.google.com/a/bioinformatics.ca/file/d/1xeYxecTL4ZtPGPjhPFOKy68Us3c2RtJQ/view?usp=sharing)  

* We have set up 30 instances on the Amazon cloud - one for each student. In order to log in to your instance, you will need a security certificate. If you plan on using Linux or Mac OS X, please download [this certificate](http://main.oicrcbw.ca/private/CBWNY.pem). Otherwise if you plan on using Windows (with Putty and Winscp), please download [this certificate](http://main.oicrcbw.ca/private/CBWNY.ppk).

* Detail instructions can be found [here](https://bioinformaticsdotca.github.io/AWS_setup).

[Lab Practical](https://bioinformaticsdotca.github.io/CSHL_2019_Module3_lab)

***

# Day 2 <a id="day2"></a>

## Module 4: Small-Variant Calling and Annotation

*<font color="#827e9c">Mathieu Bourgey</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1zavCcLwVnzmB-4bbJFjwBySsaltjtopq/view?usp=sharing)

[Lab Practical](https://bioinformaticsdotca.github.io/CSHL_2019_Module4_lab)

## Module 5: Structural Variant Calling

*<font color="#827e9c">Mathieu Bourgey</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/19zaC0qHw_H2tH41b06iyiy05Nrq5Fijo/view?usp=sharing)

[Lab Practical](https://bioinformaticsdotca.github.io/CSHL_2019_Module5_lab)   

## Module 6: De Novo Assembly

*<font color="#827e9c">Jared Simpson</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1dAnxTnVsMxpS_nQysZP3Lmq-SymP1Dc0/view?usp=sharing)

[Lab Practical](https://bioinformaticsdotca.github.io/CSHL_2019_Module6_lab)

[Supplemental](https://bioinformaticsdotca.github.io/CSHL_2019_Module6_lab_supplemental)

## Integrated Assignment - Day 2

*<font color="827e9c">Heather Gibling</font>*

[Integrated Assignment]()

***

# Day 3 <a id="day3"></a>

**Day 3 and Day 4 refer to content in the [RNA-Seq wiki](https://github.com/griffithlab/rnaseq_tutorial/wiki).**

## Module 7: Introduction to RNA Sequencing Analysis

*<font color="#827e9c">Malachi Griffith</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1blCvrQ2EJ_nPkpDfuWLzALdskd43-A2B/view?usp=sharing)

## Module 7 Lab Practical  

*<font color="#827e9c">Malachi Griffith</font>*

  [Installation](https://rnabio.org/module-00-setup/0000/08/01/Installation/)
  
  [Reference Genomes](https://rnabio.org/module-01-inputs/0001/02/01/Reference_Genomes/)
  
  [Annotations](https://rnabio.org/module-01-inputs/0001/03/01/Annotations/)
  
  [Indexing](https://rnabio.org/module-01-inputs/0001/04/01/Indexing/)
  
  [Data](https://rnabio.org/module-01-inputs/0001/05/01/RNAseq_Data/)
  
  [Data QC](https://rnabio.org/module-01-inputs/0001/06/01/Pre-alignment_QC/)


## Module 8: RNA-seq Alignment and Visualization

*<font color="#827e9c">Christopher Miller</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/11mznj7y9IbV7sOnihZAIyYv-nwHZNWmk/view?usp=sharing)

## Module 8 Lab Practical  

*<font color="#827e9c">Christopher Miller</font>*

  [Adapter Trim](https://rnabio.org/module-02-alignment/0002/02/01/Adapter_Trim/)
  
  [Alignment](https://rnabio.org/module-02-alignment/0002/03/01/Alignment/)

  [Alignment Visualization](https://rnabio.org/module-02-alignment/0002/04/01/Alignment_Visualization/)
  
  [Alignment QC](https://rnabio.org/module-02-alignment/0002/05/01/Alignment_QC/)
  
  [IGV](https://rnabio.org/module-02-alignment/0002/06/01/IGV/)



## Integrated Assignment - Day 3

*<font color="827e9c">Kelsy Cotto</font>*

Paper: [Recurrent chimeric RNAs enriched in human prostate cancer identified by deep sequencing](http://www.ncbi.nlm.nih.gov/pubmed/21571633)

[Assignment Questions](https://rnabio.org/module-08-appendix/0008/06/01/Integrated_Assignment/)

[Assignment Answers](https://rnabio.org/module-08-appendix/0008/07/01/Integrated_Assignment_Answers/)  

***

# Day 4 <a id="day4"></a>

## Module 9: Expression and Differential Expression

*<font color="#827e9c">Christopher Miller</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1NWq-8OupdSDoN-ZNaKc6B80d2XrS9nwD/view?usp=sharing)

## Module 9 Lab Practical  

*<font color="#827e9c">Christopher Miller</font>*

 [Expression](https://rnabio.org/module-03-expression/0003/02/01/Expression/)
 
 [Differential Expression](https://rnabio.org/module-03-expression/0003/03/01/Differential_Expression/)
 
 [DE Visualization](https://rnabio.org/module-03-expression/0003/04/01/DE_Visualization/)

## Module 10: Reference Free Alignment

*<font color="#827e9c">Malachi Griffith</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1rGHVI0-ev_LQxCKs5UcD57pEfWJyD67X/view?usp=sharing)

## Module 10 Lab Practical  

*<font color="#827e9c">Malachi Griffith</font>*

[Kallisto](https://rnabio.org/module-04-kallisto/0004/02/01/Alignment_Free_Kallisto/)  

## Module 11: Isoform Discovery and Alternative Expression

*<font color="#827e9c">Kelsy Cotto</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1mn1dp2YX4hrSkDpo6BPzGca84In5CEMC/view?usp=sharing)

## Module 11 Lab Practical  

*<font color="#827e9c">Kelsy Cotto</font>*

  [Ref Guided](https://rnabio.org/module-05-isoforms/0005/02/01/Reference_Guided_Transcript_Assembly/)
  
  [De novo](https://rnabio.org/module-05-isoforms/0005/03/01/De_Novo_Transcript/)
  
  [Merging](https://rnabio.org/module-05-isoforms/0005/04/01/Transcript_Assembly_Merge/)
  
  [Differential Splicing](https://rnabio.org/module-05-isoforms/0005/05/01/Differential-Splicing/)
  
  [Splicing Visualization](https://rnabio.org/module-05-isoforms/0005/06/01/Transcript_Assembly_Visualization/)

## Integrated Assignment - Day 4

*<font color="827e9c">Kelsy Cotto</font>*

[Assignment Questions](https://rnabio.org/module-08-appendix/0008/06/01/Integrated_Assignment/)  

[Assignment Answers](https://rnabio.org/module-08-appendix/0008/07/01/Integrated_Assignment_Answers/)  

## Keeping Up-to-date with RNA-Seq Analysis Developments

For additional resources, tutorials, future directions, and more please refer to the [RNA-seq tutorial](https://rnabio.org/)

***

# Day 5 <a id="day5"></a>

## Module 12: Introduction to Pathway and Network Analysis

*<font color="#827e9c">Jüri Reimand</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1g6u3Eo_aa3LnVlOFSrXHItE5ONGKlcui/view?usp=sharing)  


## Module 13: Finding Over-Represented Pathways

*<font color="#827e9c">Jüri Reimand</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1bmdpYinb52Ok3Wq1xvzOm-iRQ0nAhg4d/view?usp=sharing)  


## Module 13 Lab Practical: Enrichment-Based Analysis - Performing ORA  

*<font color="#827e9c">Jüri Reimand and Veronique Voisin</font>*

[Lab Practical](https://bioinformaticsdotca.github.io/CSHL_2019/Module13/HT-Biology_CSHL2019_Module13_Lab.pdf)

[Genelist_GBM.txt](https://bioinformaticsdotca.github.io/CSHL_2019/Module13/Genelist_GBM.txt)  
  
[Genelist_KIRC.txt](https://bioinformaticsdotca.github.io/CSHL_2019/Module13/Genelist_KIRC.txt)   

[hsapiens.pathways.NAME.gmt](https://bioinformaticsdotca.github.io/CSHL_2019/Module13/hsapiens.pathways.NAME.gmt) 


## Module 14: Network Visualization and Analysis with Cytoscape and Reactome  

*<font color="#827e9c">Robin Haw</font>* 

[Lecture Part 1](https://drive.google.com/a/bioinformatics.ca/file/d/1jwIGC_5ACs7aHBJdsPBXHeuqlJ6JJRmS/view?usp=sharing)    


## Module 14 Lab Practical: De novo subnetwork clustering analysis - Reactome  

*<font color="#827e9c">Robin Haw</font>*

[Lab Practical](https://drive.google.com/a/bioinformatics.ca/file/d/1M28EnvgxgZG1FW9Wc2UgH4TKwzerd-Gm/view?usp=sharing)   


## Integrated Assignment - Day 5  

*<font color="#827e9c">Veronique Voisin</font>* 

[Slides](https://drive.google.com/a/bioinformatics.ca/file/d/1d3QlHGRvEs5Q8tarFYqd_xKbcaYjA5Z3/view?usp=sharing)

[Integrated Assignment]()

[Integrated Assignment Answers]()

***

# Day 6 <a id="day6"></a>

## Module 15: More Depth on Network and Pathway Analysis and Cytoscape Enrichment map

*<font color="#827e9c">Veronique Voisin</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1h6aGFK9YNRDRz6ar5QL59-uI4bXtsSVK/view?usp=sharing)  


## Module 15 Lab Practical: Cytoscape-Enrichment Map 

*<font color="#827e9c">Veronique Voisin</font>* 

[Lab Practical - Part 1](https://bioinformaticsdotca.github.io/CSHL_2019_Module15_lab_1) 

[Lab Practical - Part 2](https://bioinformaticsdotca.github.io/CSHL_2019_Module15_lab_2)    


## Module 16: Gene Function Prediction

*<font color="#827e9c">Quaid Morris</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1ZAjWBteahjXpENhZIi2HySWOCzwLp-n5/view?usp=sharing)   

[Lab Practical](https://bioinformaticsdotca.github.io/CSHL_2019_Module16_lab)  

***

# Day 7 <a id="day7"></a>

## Module 17: Regulatory Network Analysis

*<font color="#827e9c">Michael Hoffman</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1EvvRPFQUukqoNBH_6jLlOVaCyk4_RrzY/view?usp=sharing)  


## Module 17 Lab Practical: iRegulon

*<font color="#827e9c">Veronique Voisin</font>*

[Lab lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1oVpBFW845FoGJ4Ox6tZV6d9cnIyk5k-M/view?usp=sharing) 

[Lab practical](https://bioinformaticsdotca.github.io/CSHL_2019_Module17_lab)  



[Precomputed results cys file]() and [Precomputed results irf file]()  

To use the precomputed results:  

1) launch Cytoscape 

2) open the "prostate_cancer_genemania_network.cys" file 

3) go to App > iRegulon > 'Load results from the iregulon_results.irf file'  
  
[Download a zip file of the git repo here]()   


## Integrated Assignment - Day 7  

*<font color="#827e9c">Veronique Voisin</font>* 

[Slides]()

[Integrated Assignment](https://bioinformaticsdotca.github.io/CSHL_2019_IA_Day5)

[Integrated Assignment Answers](https://bioinformaticsdotca.github.io/CSHL_2019_IA_Answers_Day5)




